client/executor/wasmi/src/lib.rs

```diff
@@ -574,7 +560,12 @@ impl WasmiInstance {
 			self.allow_missing_func_imports,
 			self.missing_functions.clone(),
 			allocation_stats,
-		)
+		);
+
+		// If we couldn't unmap it, erase the memory.
```

Comment:
I don't understand this comment. And dq: what is the point of having memory_zeored?

Reply:
I leave the answer to @bkchr but probably is to erase the program bss? I.e. global varables are assumed to have zero value on program entry.

Reply:
(For the next program that will run)

Reply:
Every call into the runtime needs to start with cleared memory so that one invocation can't accidentally depend on the state from the previous invocation. (We didn't always do it and in fact we used not to clear the memory, but this lead to horrible bugs [which I had to investigate :P].)